.gitlab-ci.yml

```diff
@@ -0,0 +1,31 @@
+stages:
+  - pre 
+  - build
+
+.go-cache: &go-cache
+  key: datadog-lambda-js-go-cache
+  paths:
+    - $CI_PROJECT_DIR/.go-cache/pkg/mod
+  policy: pull
+
+generator:
+  stage: pre
+  image: registry.ddbuild.io/images/mirror/golang:alpine
+  tags: ["arch:amd64"]
+  cache: *go-cache
+  script:
+    - go install github.com/hairyhenderson/gomplate/v4/cmd/gomplate@latest
+    - gomplate --config ci/config.yaml
+  artifacts:
+    paths:
+      - ci/*-pipeline.yaml
+
+build-layers:
+  stage: build
+  trigger:
+    include:
+      - artifact: ci/build-pipeline.yaml
+        job: generator
+    strategy: depend
+  rules:
+    - when: on_success
```

Comment:
Just a question for learning purpose...
So this reads " if my build was success, go on and generate the pipelines based on the template" ?